.buildkite/windows.ps1

```diff
@@ -14,38 +14,50 @@ $env:CHECKOUT_PATH=(Get-Item -Path ".\").FullName
 # Temp directory needs to exist
 New-Item -ItemType Directory -Force -Path $env:TMP
 
-#  Setup stuff (commented out)
-#mkdir d:\ghc
+# Make sure ghc directory exists
+New-Item -ItemType Directory -Force -Path d:\ghc
 #
-#Invoke-WebRequest "https://s3.eu-central-1.amazonaws.com/ci-static/ghc-8.2.2-x86_64-unknown-mingw32.tar.xz" -OutFile "D:\ghc\ghc.tar.xz" -UserAgent "Curl"
-#
-#7z x D:\ghc\ghc.tar.xz -oD:\ghc
-#
-#7z x D:\ghc\ghc.tar -oD:\ghc
+if (!([System.IO.File]::Exists("D:\ghc\ghc-8.2.2.tar.xz"))) {
+  echo "Downloading and extracting GHC"
+  curl.exe https://s3.eu-central-1.amazonaws.com/ci-static/ghc-8.2.2-x86_64-unknown-mingw32.tar.xz -o D:\ghc\ghc-8.2.2.tar.xz -L
+  7z x D:\ghc\ghc-8.2.2.tar.xz -oD:\ghc
+  7z x D:\ghc\ghc-8.2.2.tar -oD:\ghc
+}
 
 # OpenSSL
 #
 #$env:USERPROFILE
-#(New-Object Net.WebClient).DownloadFile('https://slproweb.com/download/Win64OpenSSL-1_0_2p.exe', "D:\Downloads\Win64OpenSSL.exe")
-#cmd /c start /wait "D:\Downloads\Win64OpenSSL.exe" /silent /verysilent /sp- /suppressmsgboxes /DIR=D:\OpenSSL-Win64-v102
+if (!([System.IO.File]::Exists("D:\Downloads\Win64OpenSSL.exe"))) {
+  echo "Downloading and installing OpenSSL"
+  curl.exe https://slproweb.com/download/Win64OpenSSL-1_0_2q.exe -o D:\Downloads\Win64OpenSSL.exe -L
+  cmd /c start /wait "D:\Downloads\Win64OpenSSL.exe" /silent /verysilent /sp- /suppressmsgboxes /DIR=D:\OpenSSL-Win64-v102
+}
 ## Install stack
-#Start-FileDownload http://www.stackage.org/stack/windows-x86_64 -FileName D:\Downloads\stack.zip
-#curl.exe http://www.stackage.org/stack/windows-x86_64 -o d:\Downloads\stack.zip -L
-#7z -oD:\stack x D:\Downloads\stack.zip
+if (!([System.IO.File]::Exists("D:\Downloads\stack.zip"))) {
+  echo "Downloading and extracting stack"
+  curl.exe http://www.stackage.org/stack/windows-x86_64 -o d:\Downloads\stack.zip -L
+  7z -oD:\stack x D:\Downloads\stack.zip
+}
 
 $env:PATH="$env:PATH;D:\ghc\ghc-8.2.2\bin;D:\stack;$Env:Programfiles\7-Zip;D:\w"
```

Comment:
7z is already in the PATH by this point (it was used 3 lines ago), this is likely to duplicate the 7z's in PATH